nautobot/core/graphql/generators.py

```diff
@@ -108,15 +112,23 @@ def generate_list_search_parameters(schema_type):
     """Generate list of query parameters for the list resolver based on a filterset."""
 
     search_params = {}
-    exclude_filters = ["type"]
-    if schema_type._meta.filterset_class:
-        for key in list(schema_type._meta.filterset_class.get_filters().keys()):
-            if key in exclude_filters:
-                continue
-            if key == "id":
-                search_params[key] = graphene.ID()
-            else:
-                search_params[key] = graphene.String()
+    if schema_type._meta.filterset_class is not None:
+        # We need an instance for custom fields generation to happen for the
+        # filterset_class or the `cf_*` fields won't be detected.
+        filterset = schema_type._meta.filterset_class()
+        # Patch base_filters because `get_filtering_args_from_filterset` looks there
+        filterset.base_filters = filterset.filters
+
+        search_params = get_filtering_args_from_filterset(
+            filterset,
+            schema_type,
+        )
+
+    # Hack to swap `type` fields to `type_` since they will conflict with
+    # `graphene.types.fields.Field.type`.
+    # FIXME(jathan): Once we upgrade to Graphene 3.x, remove this.
+    if "type" in search_params:
+        search_params["type_"] = search_params.pop("type")
```

Comment:
Do I understand correctly that Graphene 3.x reverses this (i.e., `type` is a valid variable name but `type_` will conflict)? If so, should we pick a name (such as `_type` or `type__`) that will work as-is regardless of the Graphene version? Trying to reduce our need for breaking API changes in the future. :-)

Reply:
It's both. It ends up being stored as `self._type` on the instance. Both of them just unpack kwargs (as `**extra_args`, and stores them on `self.args`.

- Graphene v2.1.8 (what we are currently using) [uses `type` as the argument](https://github.com/graphql-python/graphene/blob/v2.1.8/graphene/types/field.py#L65)
- Graphene v3.x uses [`type_` as the argument](https://github.com/graphql-python/graphene/blob/v3.0.0b7/graphene/types/field.py#L68)

The other option is to just upgrade to Graphene 3.0.0b7+ now and cut out the middleman. I attempted to do that, but opened up a rabbit hole on an API change in `graphql-core` that no longer requires GraphQL "backends" to be defined. (We're using `from graphql import get_default_backend`) in [`nautobot.core.api.views.GraphQLDRFAPIView`](https://github.com/nautobot/nautobot/blob/develop/nautobot/core/api/views.py#L360)). It wasn't a quick win, but might be worth it just to future proof so we could just roll forward with `type` unhindered.

Reply:
I revisited this again to try to upgrade to Graphene 3.x and failed, but I did validate that renaming to `_type` will be safe for both 2.x and 3.x, so I will swap that.